src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -487,5 +487,59 @@ private static ImmutableArray<ISymbol> GetOverriddenSymbols(ISymbol memberSymbol
                     cancellationToken: cancellationToken).ConfigureAwait(false);
             }
         }
+
+        /// <summary>
+        /// Create the DefinitionItem based on the numbers of locations for <paramref name="symbol"/>.
+        /// If there is only one location, create the DefinitionItem contains only the documentSpan or symbolKey to save memory.
+        /// Because in such case, when clicking InheritanceMarginGlpph, it will directly navigate to the symbol.
+        /// Otherwise, create the full non-classified DefinitionItem. Because in such case we want to display all the locations to the user
+        /// by reusing the FAR window.
+        /// </summary>
+        private static async Task<DefinitionItem> ToSlimDefinitionItemAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
+        {
+            var locations = symbol.Locations;
+            if (locations.IsEmpty)
+            {
+                return DefinitionItem.CreateNonNavigableItem(
+                    tags: GlyphTags.GetTags(symbol.GetGlyph()),
+                    displayParts: FindUsagesHelpers.GetDisplayParts(symbol));
```

Comment:
note: since we can't anvigate... is there any point to returning this item?  consider TryCreateSlimDefinitionItem that can return null, and have the caller filter those out.

Reply:
Done